tests/posts/test_posts_list_endpoint.py

```diff
@@ -0,0 +1,266 @@
+
+    result = client.get("api/topics/1/posts", headers={"Authorization": f"Bearer {authorization_token}"},
+                        query_string={"created_after": "2023-05-07 00:12:38.078953"})
+
+    assert result.status_code == 200
+    assert result.json == [
+        {
+            "id": 3,
+            "author_id": 3,
+            "body": "test body3",
+            "created_at": "2023-05-08 00:12:38.078953",
+            "topic_id": 1,
+        },
+    ]
+
+
+def test_get_posts_list_endpoint_with_empty_created_before_returns_400_with_correct_body(client, db_with_one_user):
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.get("api/topics/1/posts", headers={"Authorization": f"Bearer {authorization_token}"},
+                        query_string={"created_before": ""})
+
+    assert result.status_code == 400
+    assert result.json == {"message": "empty argument value is not allowed"}
+
+
+def test_get_posts_list_endpoint_with_invalid_created_before_returns_400_with_correct_body(client, db_with_one_user):
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.get("api/topics/1/posts", headers={"Authorization": f"Bearer {authorization_token}"},
+                        query_string={"created_before": "InvalidArgument"})
+
+    assert result.status_code == 400
+    assert result.json == {"message": "invalid datetime string"}
+
+
+def test_post_posts_list_endpoint_with_valid_args_returns_200_with_correct_body(client, db_with_one_user_and_one_topic):
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.post("api/topics/1/posts", headers={"Authorization": f"Bearer {authorization_token}"},
+                         query_string={"body": "test"})
+
+    assert result.status_code == 200
+    response_body = result.json
+    post_id = response_body.pop("id")
+    assert isinstance(post_id, int)
+    assert response_body == {
+        # "id": 10000,  # noqa
+        "author_id": 1,
+        "body": "test",
+        "topic_id": 1,
+    }
+
+
+def test_post_posts_list_endpoint_with_valid_args_creates_objects_in_db_correctly(
+        client, db_with_one_user_and_one_topic,
+):
+    session = db_with_one_user_and_one_topic
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.post("api/topics/1/posts", headers={"Authorization": f"Bearer {authorization_token}"},
+                         query_string={"body": "test"})
+
+    posts = session.query(Post).all()
+    assert len(posts) == 1
+    created_post = posts[0]
+    assert created_post.created_at < datetime.now()
+    assert created_post.body == "test"
+    assert created_post.author_id == 1
+    assert isinstance(created_post.id, int)
+
+
+def test_post_posts_list_endpoint_with_nonexistent_topic_returns_404_with_correct_body(client,
```

Comment:
This test returns `400` when it's run together with other tests, because of the parser. Now we create one parser instance for all routes within a file. And after that each route adds each own arguments to the parser.

This means that we run all tests for routes within a file the parser is configured by two routes and has arguments for two routes, but we need a parser to have arguments only for one route each time.

This should be fixed by creating a separate parser instance for each route. Created a [ticket](https://github.com/fenya123/forum123/issues/154) for that.